### PR TITLE
add q-Fuss-Catalan numbers

### PR DESCRIPTION
**Pull request by fchapoton**
This is adding the q-Fuss-Catalan numbers among the q-analogues.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/q_analogues.py [hunk 1]
```diff
@@ -452,10 +451,16 @@ def q_multinomial(seq, q=None, binomial_algorithm='auto'):
...
 
 
-def q_catalan_number(n, q=None):
+def q_catalan_number(n, m=1, q=None):
     """
     Return the `q`-Catalan number of index `n`.
 
+    INPUT:
+
+    - ``m`` -- optional integer (default: 1) to get the ``m``-Fuss-Catalan numbers
```
Comment (tscrim) [inline src/sage/combinat/q_analogues.py]:
```suggestion
    - ``m`` -- (optional) the ``m``-Fuss-Catalan number
```

File: src/sage/combinat/q_analogues.py [hunk 2]
```diff
@@ -866,7 +877,7 @@ def q_stirling_number1(n, k, q=None):
         q^3 + 3*q^2 + 4*q + 3
 
         sage: all(stirling_number1(6,k) == q_stirling_number1(6,k)(1)
-        ....:     for k in range(1,7))
+        ....:     for k in range(1,6))
```
Comment (tscrim) [inline src/sage/combinat/q_analogues.py:880]: Why this change?

File: src/sage/combinat/q_analogues.py [hunk 3]
```diff
@@ -467,10 +472,16 @@ def q_catalan_number(n, q=None):
         sage: from sage.combinat.q_analogues import q_catalan_number
         sage: q_catalan_number(4)
         q^12 + q^10 + q^9 + 2*q^8 + q^7 + 2*q^6 + q^5 + 2*q^4 + q^3 + q^2 + 1
+
         sage: p = ZZ['p'].0
-        sage: q_catalan_number(4,p)
+        sage: q_catalan_number(4,q=p)
```
Comment (tscrim) [inline src/sage/combinat/q_analogues.py]:
```suggestion
        sage: q_catalan_number(4, q=p)
```
That this change is needed makes it clear we are breaking backwards compatibility.

**Review by tscrim — CHANGES_REQUESTED** [review]
Overall this is fine, but I don't like breaking the backwards compatibility of the argument order. There isn't a very good way out of this, but we could at least catch the most obvious cases by
```python
if m in ZZ`:
    if q is not None:
        raise ValueError(f"{m} must be an integer")
    q = m
    deprecation(36314, f"the second argument has changed to m, use 'q={q}' instead)
```

**Comment by fchapoton** [comment]
Thanks. I have now restored backward compatibility.

File: src/sage/combinat/q_analogues.py (hunk 2 above)
Comment (fchapoton) [reply src/sage/combinat/q_analogues.py:880]: this was slightly slow and rather superfluous

File: src/sage/combinat/q_analogues.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/combinat/q_analogues.py]: not sure I can agree with this suggestion, so I did not perform that

**Comment by mantepse** [comment]
I agree that moving the `m` to the end is better.

**Review by mantepse — APPROVED** [review]
LGTM

File: src/sage/combinat/q_analogues.py (hunk 1 above)
Comment (tscrim) [reply src/sage/combinat/q_analogues.py]: An optional argument should not have a default; in terms of the documentation that is. If it has a default, then it is mandatory. Since this is treated as something optional for the user to input, I think it should be marked as such. If we want to say it is a default (and hence essentially required), then I think we should document it as such. Do you agree or have an argument against this?

**Comment by fchapoton** [comment]
ok, I agree. Here is a new tentative for the description of the optional parameter m.

File: src/sage/combinat/q_analogues.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/combinat/q_analogues.py]: ok, now I understand your point. I made a new tentative

**Review by tscrim — APPROVED** [review]
Thanks.
